<launch>

  <arg name="robot_name" default="robot_0" />

  <group ns="$(arg robot_name)">
    <param name="tf_prefix" value="$(arg robot_name)" />

    <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <param name="controller_frequency" value="10.0" />
      <rosparam file="$(find my_patrol_sim)/param_sim/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find my_patrol_sim)/param_sim/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find my_patrol_sim)/param_sim/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find my_patrol_sim)/param_sim/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find my_patrol_sim)/param_sim/base_local_planner_params.yaml" command="load" />
      <rosparam file="$(find my_patrol_sim)/param_sim/move_base_params.yaml" command="load" />

      <param name="global_costmap/global_frame" value="map" />
      <param name="global_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint" />
      <param name="local_costmap/global_frame" value="$(arg robot_name)/odom" />
      <param name="local_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint" />
      <param name="DWAPlannerROS/global_frame_id" value="$(arg robot_name)/odom" />

    </node>

    <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
      <param name="odom_frame_id" value="$(arg robot_name)/odom" />
      <param name="base_frame_id" value="$(arg robot_name)/base_footprint" />
    </node>

  </group>

</launch>
